src/compat/posix/proc/fork_nommu/fork_addr_space_heap.c

```diff
@@ -28,20 +29,33 @@ static inline struct dlist_head *task_mspace(struct task *tk) {
 void fork_heap_store(struct heap_space *hpspc, struct task *tk) {
 	size_t size;
 
+	assert(hpspc);
+	assert(tk);
+
 	size = mspace_deep_copy_size(task_mspace(tk));
+	if (!size) {
+		return;
+	}
 
 	if (hpspc->heap_sz != size) {
 		if (hpspc->heap) {
 			phymem_free(hpspc->heap, hpspc->heap_sz / PAGE_SIZE());
 		}
 
 		hpspc->heap = phymem_alloc(size / PAGE_SIZE());
-		assert(hpspc->heap);
+		hpspc->heap_sz = size;
 	}
+	assert(hpspc->heap);
 	mspace_deep_store(task_mspace(tk), &hpspc->store_space, hpspc->heap);
 }
 
 void fork_heap_restore(struct heap_space *hpspc) {
+	assert(hpspc);
+
+	if (NULL == hpspc->heap) {
+		return;
```

Comment:
Why could it happen? Is it correct situation?

Reply:
it's happend if we is the commetn case
```case  in +	if (!size) {
	return;```